src/pages/OrderDetail/OrderDetail.tsx

```diff
@@ -1,11 +1,37 @@
-import React, { FC } from "react";
+import React, { FC, useEffect } from "react";
+import { useParams } from "react-router";
 
+import { useAppDispatch, useAppSelector } from "../../app/hooks";
 import OrderDetailWidget from "../../components/OrderDetailWidget/OrderDetailWidget";
 import Page from "../../components/Page/Page";
-
-// swap detail page (swap-id/:datauri)
+import { decompressAndSetActiveOrder } from "../../features/takeOtc/takeOtcActions";
+import { selectTakeOtcReducer } from "../../features/takeOtc/takeOtcSlice";
+import InvalidOrder from "./subcomponents/InvalidOrder/InvalidOrder";
 
 const OrderDetail: FC = () => {
+  const dispatch = useAppDispatch();
+  const { compressedOrder } = useParams<{ compressedOrder: string }>();
+
+  const { status } = useAppSelector(selectTakeOtcReducer);
+
+  useEffect(() => {
+    if (compressedOrder) {
+      dispatch(decompressAndSetActiveOrder(compressedOrder));
```

Comment:
```suggestion
      dispatch<any>(decompressAndSetActiveOrder(compressedOrder));
```

probably a better fix, but I get a type/overload error here and I used this to stop it

Reply:
Strange, I don't get an error. Will add to fix things on your end.